### PR TITLE
Fixed #26889 -- Fixed missing PostgreSQL index in SchemaEditor.add_field().

### PR DESCRIPTION
**Pull request by jdufresne**
<https://code.djangoproject.com/ticket/26889>

### CODE REVIEW

**Comment by timgraham** [comment]
If we apply the similar reasoning as ticket #25412, we might consider this a data loss issue worthy of backporting to 1.8/1.9. Any thoughts?  In that case, I wonder if the schema editor changes might be a bit too invasive for a stable release (concerned about breaking a third-party database backend).

**Comment by jdufresne** [comment]
I have created an alternative version of the patch that is less invasive and only involves the PostgreSQL backend. You can view it be viewing the full diff, not just the last commit. If you prefer the 2nd version I will squash the commits.

I think I like the original better as it offers a single hook to backends to inject indexes on fields (instead of adding the same index in multiple places). But ultimately both work and I am happy with either. I do see your point wrt backporting the change, so whatever makes that work easier is no problem.

Thanks. All feedback welcome.

**Comment by timgraham** [comment]
Looks good. We can merge this approach and then follow up with a refactoring on master for the other approach. I'm not sure the `SlugField` test is needed (or all other subclasses of `CharField`).

File: tests/schema/tests.py [hunk 1]
```diff
@@ -1908,6 +1908,45 @@ def test_add_textfield_unhashable_default(self):
             editor.add_field(Author, new_field)
 
     @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific")
+    def test_add_field_with_index_with_charfield(self):
```
Comment (charettes) [inline tests/schema/tests.py]: `test_add_charfield_with_index` seems like a more appropriate name and follows the format used in previous test methods.

File: tests/schema/tests.py [hunk 2]
```diff
@@ -1908,6 +1908,45 @@ def test_add_textfield_unhashable_default(self):
...
+            editor.create_model(Author)
+            editor.add_field(Author, field)
+        # Should create two indexes; one for like operator.
+        self.assertEqual(
+            self.get_constraints_for_column(Author, 'nom_de_plume'),
+            ['schema_author_95aa9e9b', 'schema_author_nom_de_plume_7570a851_like'],
+        )
+
+    @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific")
+    def test_add_field_with_unique_with_charfield(self):
```
Comment (charettes) [inline tests/schema/tests.py]: Ditto with `test_add_charfield_with_unique`.

**Comment by jdufresne** [comment]
Made requested changes and squashed commits. Thanks.

**Pull request merged**